src/index.ts

```diff
@@ -120,34 +115,33 @@ export class GCEImages {
    * @param {boolean} options.deprecated [false] - Include deprecated results.
    * @param {array} options.osNames [all] - OS names to include in the results.
    * @param {function} callback - Callback function.
+   * @returns {Promise} if callback is omitted.
    */
-  getAll(cb: GetAllCallback): void;
-  getAll(opts: GetOptions|string, cb: GetAllCallback): void;
-  getAll(opts: GetOptions|string|GetAllCallback, cb?: GetAllCallback): void {
+  async getAll(cb: GetAllCallback): Promise<void>;
+  async getAll(opts?: GetOptions|string): Promise<Image[]|ImagesMap>;
+  async getAll(opts: GetOptions|string, cb: GetAllCallback): Promise<void>;
+  async getAll(
```

Comment:
I don't think that's going to work.  If you flag the method as async, it will always create a wrapping promise.  What you need to do is create a protected async method, then call it from a non-async method.  For an example of this pattern see:
https://github.com/googleapis/google-auth-library-nodejs/blob/master/src/auth/googleauth.ts#L193

Reply:
Refactored accordingly, thanks.